ast/const.go

```diff
@@ -1,5 +1,14 @@
 package ast
 
+type AllOrDistinct string
+
+const (
+	// AllOrDistinctUnspecified is zero value of AllOrDistinct, and it must be ignored in SQL().
```

Comment:
Constant names must be `All` and `Distinct`, and zero value is unnecessary.

Reply:
I think `All` is too general a name to bind only `AllOrDistinct`.
If so, how would you name `ALL` in the GQL path search prefix?

https://cloud.google.com/spanner/docs/reference/standard-sql/graph-patterns#search_prefix

```
path_search_prefix:
  {
    ALL
    | ANY
    | ANY SHORTEST
  }
```

Reply:
I'm looking for external guidelines for constants to reference, but I can't find much.
https://google.github.io/styleguide/go/decisions.html#constant-names

> Name constants based on their role, not their values. If a constant does not have a role apart from its value, then it is unnecessary to define it as a constant.
> 
> ```
> // Bad:
> const Twelve = 12
> 
> const (
>     UserNameColumn = "username"
>     GroupColumn    = "group"
> )
> ```

Reply:
Okay, `AllOrDistinctAll` and `AllOrDistinctDistinct` is accepted.